dockertest/subtest.py

```diff
@@ -429,13 +458,62 @@ def try_all_stages(self, name, subsubtest):
                               self.exception_info["error_source"],
                               detail)
             exc_info = self.exception_info["exc_info"]
+            # Ignore known problems in specific docker builds
+            if self.is_known_failure(name):
+                self.final_subsubtests.add(name)
             # Treat N/A as passed, even though this may hide obscure failures
             if isinstance(detail, TestNAError):
                 self.logwarning("Treating TestNAError as PASS")
                 self.final_subsubtests.add(name)
             # cleanup() will still be called before this propagates
             raise exc_info[0], exc_info[1], exc_info[2]
 
+    def is_known_failure(self, subsubtestname):
+        """
+        Do we have a registered known failure in this subtest when running
+        on the currently-installed docker? Return True if so.
+        Side effect: log warning messages to help human debuggers.
+        """
+        # e.g. docker_cli/subtest/subsubtest
+        fullname = os.path.join(self.config_section, subsubtestname)
+        known = known_failures()
```

Comment:
> Clarification: is_known_failure only gets called on failure. There is no file reading, no performance penalty, on successful completion of a test.

Oooohhhh, okay, my bad again.  I thought it was being called after every single one.  Errr, wait..no, it's github's fault!  I just looked, it did some code/diff-folding that totally covered up:

```python
 except Exception, detail:
    ...
```

right before ``if self.is_known_failure(name):``.  Though I suppose it's my fault for assuming the scope.  In that case, what you have here is fine.  No caching needed.
```